watsononlinestore/watson_online_store.py

```diff
@@ -292,6 +294,23 @@ def get_discovery_response(self, input_text):
             query_options={'query': input_text, 'count': DISCOVERY_QUERY_COUNT}
         )
 
+        # Watson discovery assigns a confidence level to each result.
+        # Based on data mix, we can assign a minimum tolerance value in an
+        # attempt to filter out the "weakest" results.
+        num_matches = 0
+        filtered_results = []
+        if self.discovery_score_filter:
+            results = (results for results in discovery_response['results']
+                      if 'results' in discovery_response)
+            for result in results: 
+                if ('score' in result and
+                   result['score'] >= self.discovery_score_filter):
+                    filtered_results.append(result)
+                    num_matches += 1
+
```

Comment:
I must have made a cut/paste error initially when a git issue required me to recreate this patch. The following 2 lines should be indented, and defs for num_matches and filtered_results should be moved inside this loop. We don't want to rebuild the discovery_response if no score filter is specified.